### PR TITLE
Alternate bionic system mod 

### PR DESCRIPTION
**Pull request by John-Candlebury**
An alternate system for the installation of CBMs. 
Simmilar to the system in Caves of Qud, it follows the outline I generally described in #24400

It works as follows:
- The player is given an additional limitation on installation, the bionic license.  The license can be upgraded at an autodoc through the use of license cards, which can be found in several science-y locations. 
- CBMs are given a license cost. Power storage and faulty bionics have no license cost. All other bionics have a license cost of 2. 
- Professions start with a license score equal to the cost of all their bionics. This greatly buffs all bionic professions.
- You cannot install bionics if the  cost of installed bionics+cost of the new bionic exceeds the value of your bionic license.
- You can freely and safely swap bionics at an autodoc. Allowing you to greatly customize your bionics to your immediate needs.
- Anesthesia requirement is removed to actually allow the free swapping to occur. 

Should I close #24408 and implement it  as game option?

### CODE REVIEW

**Comment by Night-Pryanik** [comment]
This will interfere with manual installation. How is remaining license score could be potentially checked on manual installation?

Overall I don't like it. Of course, it's a mod, but nevertheless. Autodoc is a more or less sterile robotic surgeon, not a some sort of ATM that sells licenses, accepting undoubtedly dirty plastic cards. Autodocs isn't supposed to be connected to banks to transfer them electronic money, that's just doesn't make sense. 

The whole functional of selling (upgrading) licenses should be moved to ATM, or some computer at banks, or wherever. Hospitals', clinics' or infirmaries' operational theater with autodoc is definitely not the proper place to make financial operations.

**Comment by AMurkin** [comment]
Counter = Counter + 2 not really looks as a firmware upgrade. More like as additional modules of processing units (SIM-like).

**Comment by darktoes** [comment]
You could even consider it like an actual license. Perhaps the government decided having people running rampant with godlike strength and general superpowers was a bad idea, and decided to encode bionics so they would only activate if they had a valid license attached, something sort of like cryptocurrency, but tracking who and what has the bionic installed, and from where.

You might even be able to look at black market bionics which are inferior, maybe even with special penalties and/or prevent you from using legit bionics, but don't require license points. Or ones that are horrendously overpowered, like a supercharged (highly illegal) fusion blaster arm.

**Comment by John-Candlebury** [comment]
@Night-Pryanik 

This is not really the final mod, just the basic code support that allows it to work, so that other people might use it and give me general feedback. 

When/if this get merged I'll start a forum thread to ask for feedback.

>This will interfere with manual installation. How could be remaining license score potentially checked out on manual installation?

If manual installation is on, the assumption is that CBMs  work according to the old descriptions, aka nano-tech based specialized surgery kits, they being able to check your license is not unjustified. 

The bionics themselves also checking the license after they are installed is also not unjustified.

In any case this mod doesn't allow manual installation. Adding special Illegal installations would not be out of question, but I rather have user feedback first.

>The whole functional of selling (upgrading) licenses should be moved to ATM, or some computer at banks, or wherever

Dont consider the upgrading your license like transferring money between bank accounts, your license is directly inscribed into your body and interacting with it is a complex procedure. The doc doesn't sell licenses it permanently transfers them between the cards you bought and your body.

Other fact is that letting players to everything bionic upgrade related at a single place is much less confusing and simpler for them. Replacing the auto-doc with a more general cybernetics station would not be out of question for this mod,  but that would require some map making and I'd rather keep that for later PRs. 

In any case, this is meant to be part of brightnights later on, the general flavor of brightnights is cyberpunk. Having to pay an auto-doctor before it will heal you has a strong cyberpunk feel to it.

If you strongly oppose I can introduce the new piece furniture right now, and make the mod overwrite the autodoc.

**Comment by John-Candlebury** [comment]
This is failing because a linter rule is missing. Wheres the new equivalent to the format.conf files so I might add the rule?

**Comment by Night-Pryanik** [comment]
> nano-tech based specialized surgery kits, they being able to check your license

That's too much for surgery nanobots.

> your license is directly inscribed into your body

I don't get it, that means that if character wants to install, say, 10 bionics, then he will have 10, uhmm, "tatoos" inscribed?

> If you strongly oppose I can introduce the new piece furniture right now, and make the mod overwrite the autodoc.

Yes, please. Autodoc supposed to be robotic surgeon, not a license checker of any sort.

> This is failing because a linter rule is missing.

And also because you need to astyle your changes.

**Comment by John-Candlebury** [comment]
>That's too much for surgery nanobots.

That's your opinion, so we will have to agree to disagree.

>I don't get it, that means that if character wants to install, say, 10 bionics, then he will have 10, uhmm, "tatoos" inscribed?

Thats a valid way of seeing it.  Another version would be: They have a futuristic bar code inscribed that allows them to install 10 points worth of bionics.

>Yes, please. Autodoc supposed to be robotic surgeon, not a license checker of any sort.

Done. For now this mod overwrites the normal autodoc, and couch. If this PR makes it i'll introduce bionic augmentation centers specific for this mod and remove the overwrite.

**Comment by kevingranade** [comment]
Instead of a physical card for the license, can it be assumed to be its own CBM and be a property of the player?  It doesn't have to be represented in the bionic menu or anything, just thematically.

Implementation-wise it would just be a player tag like ```"BIONIC_LICENSE": 10```

File: data/mods/Alternate_Bionic_System/flags.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+[
+  {
+    "id": "CBM",
+    "type": "json_flag",
+    "context": [ "BIONIC_ITEM" ],
+    "info": "This item is a Compact Bionic Module.  You can <info>use an autodoc</info> an <info>augmentation console</info> or <info>ask a surgeon</info> to install it into your body."
```
Comment (Night-Pryanik) [inline data/mods/Alternate_Bionic_System/flags.json]: Remove mention of autodoc.

**Comment by Mecares** [comment]
What about a bionics control unit that is a bionic itself, that is required to control your bionics by being the interface that conects the bionics to your brain and limits the amount of bionics a character can have depending on its processing power. It can be upgraded and expanded and makes more technical sense than a license tattoo as a limitation on how many bionics a character can have? It would also allow logically for some bionics to not require additional licenses like power storage.

**Comment by John-Candlebury** [comment]
Whats up with the .editorconfig file showing fake changes?

Any idea how to fix that?

**Comment by dissociativity** [comment]
I mean this could be more motivation to have VERY high computer skills and perhaps hack the autodoc to bypass the license limits, risking the thing no longer functioning on failure.

**Comment by ZhilkinSerg** [comment]
> Whats up with the .editorconfig file showing fake changes?

Probably invalid type of line break/new line character.

**Comment by ZhilkinSerg** [comment]
I've put this beyond repair while trying to resolve conflicts. Also there were no activity from you for a while.

I will close it to free PR queue. Please ping me to reopen if you wish to continue working on this PR.

**Pull request closed without merging**